app/src/main/java/com/epmedu/animeal/tabs/TextInputField.kt

```diff
@@ -0,0 +1,313 @@
+            text = annotatedString,
+            offsetMapping = object : OffsetMapping {
+                override fun originalToTransformed(offset: Int): Int {
+                    var lastIndex = 0
+                    while (true) {
+                        when {
+                            lastIndex == map.keys.size -> {
+                                return mask.length
+                            }
+                            offset <= map.keys.toList()[lastIndex] -> {
+                                return offset + lastIndex
+                            }
+                            else -> {
+                                lastIndex += 1
+                            }
+                        }
+                    }
+                }
+
+                override fun transformedToOriginal(offset: Int): Int {
+                    var lastIndex = 0
+                    while (true) {
+                        when {
+                            lastIndex == map.keys.size -> {
+                                return mask.length
+                            }
+                            offset <= map.keys.toList()[lastIndex] -> {
+                                return offset - lastIndex
+                            }
+                            else -> {
+                                lastIndex += 1
+                            }
+                        }
+                    }
+                }
+            }
+        )
+    }
+}
+
+@Preview(
+    showBackground = true,
+    name = "TextInputField",
+    showSystemUi = true,
+    uiMode = UI_MODE_NIGHT_NO,
+)
+@Composable
+fun TextInputPreview() {
+    AnimealTheme {
+        Surface(color = MaterialTheme.colors.background, modifier = Modifier.fillMaxSize()) {
+            var name by remember { mutableStateOf("") }
+
+            TextInputField(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(horizontal = 16.dp),
+                title = "Name",
+                placeholder = "Enter your name",
+                onValueChange = {
+                    name = it
+                },
+                name
+            )
+        }
+    }
+}
+
+@Preview(
+    showBackground = true,
+    name = "MaskInputField",
+    showSystemUi = true,
+    uiMode = UI_MODE_NIGHT_NO,
+)
+@Composable
+fun PhoneInputPreview() {
+    AnimealTheme {
+        Surface(
+            color = MaterialTheme.colors.background,
+            modifier = Modifier
+                .fillMaxSize()
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize(),
+                horizontalAlignment = Alignment.CenterHorizontally,
+            ) {
+
+                Spacer(modifier = Modifier.height(20.dp))
+
+                var phoneNumber by remember { mutableStateOf("") }
+                /**
+                 *  PHONE NUMBER INPUT
+                 */
+                FormattedInputView(
```

Comment:
I suggest to create `ChipNumberInput` and `PhoneNumberInput`. 
All the common logic can be shared across this two functions.

Reply:
Done

Reply:
I added separately ChipNumberInput